tracee: 0.7.0 -> 0.8.3

06kellyjac opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Updated tracee to 0.8.0. Swapped to using our copy of libbpf since it's now a matching version to the one needed by tracee


After #176152 the tracee build process can now pass in a -target of bpf
without weird overrides

the CFLAG is only needed for the bpf target but it's much easier to just apply it generally, means less maintenance going forward
I might ask upstream for more controllable setting of cflags of individual targets

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

symphorien commented on pkgs/tools/security/tracee/default.nix:
```diff
@@ -39,34 +28,31 @@ buildGoModule rec {
     "CMD_GIT=echo"
   ];
 
+  NIX_CFLAGS_COMPILE = "-fno-stack-protector";
```
```suggestion
  hardeningDisable = [ "stackprotector" ];
```

06kellyjac replied:
Thanks. I'll fix that

06kellyjac commented:
@ofborg eval

symphorien commented on pkgs/tools/security/tracee/default.nix:
```diff
@@ -6,67 +6,55 @@
   '';
 
+  # tests require a separate go module
+  # integration tests can be ran via ./test.nix in a nixos vm
```
can you add the test to passthru.tests ?

06kellyjac replied:
done

06kellyjac committed:
tracee: 0.7.0 -> 0.8.3

Also simplified the package since after #176152 the tracee build process can
now pass in a -target of bpf without weird overrides

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/1016/24

symphorien committed:
Merge branch 'master' into tracee-use-new-wrapper
